File: module-app/src/app/java/com/huyingbao/app/SimpleApplication.java
package com.huyingbao.app;


import com.huyingbao.core.CommonApp;

import dagger.android.AndroidInjector;
import dagger.android.support.DaggerApplication;

/**
 * Created by liujunfeng on 2017/12/7.
 */
public class SimpleApplication extends CommonApp {
    @Override
    protected AndroidInjector<? extends DaggerApplication> applicationInjector() {
        return DaggerSimpleComponent.builder().application(this).build();
    }
}

File: module_app_kotlin/src/app/java/com/huyingbao/module/app/SimpleApplication.kt
package com.huyingbao.module.app


import com.huyingbao.core.common.CommonApp

import dagger.android.AndroidInjector
import dagger.android.support.DaggerApplication

/**
 * 主Application
 * Created by liujunfeng on 2019/1/1.
 */
class SimpleApplication : CommonApp() {
    @Override
    protected fun applicationInjector(): AndroidInjector<out DaggerApplication> {
        return DaggerSimpleComponent.builder().application(this).build()
    }
}
